### PR TITLE
Set up form handling

### PR DESCRIPTION
**Pull request by Moomia**
- I opted to use Formik, an open-source form library for React (_reason: for days I've been working on turning our forms into controlled components, but manually trying to handle the state of each and every input field is just way too cumbersome_)
- Rewrote the existing forms using the library, now they are controlled components and ready for AJAX calls

### CODE REVIEW

**Review by Ostend — APPROVED** [review]

File: frontend/src/components/SignUp.jsx [hunk 1]
```diff
@@ -1,40 +1,44 @@
...
+import { Formik, Form, Field } from "formik";
 
 const SignUp = () => {
-  const handleSubmit = (evt) => {
-    evt.preventDefault();
-    console.log("Form submitted");
-  };
-
   return (
     <>
```
Comment (rodgobbi) [inline frontend/src/components/SignUp.jsx]: We can remove this React Fragment.

File: frontend/src/components/LogIn.jsx [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
+import { Formik, Form, Field } from "formik";
 
 const LogIn = () => {
-  const handleSubmit = (evt) => {
-    evt.preventDefault();
-    console.log("Form submitted");
-  };
-
   return (
     <>
```
Comment (rodgobbi) [inline frontend/src/components/LogIn.jsx]: We can remove the React Fragment here.

File: frontend/src/components/RecommendationForm.jsx [hunk 3]
```diff
@@ -1,88 +1,70 @@
...
-    //   setPrice(price);
-    //   if (vegan) {
-    //     setVegan(vegan);
-    //     console.log(vegan);
-    //   }
-    // }
-  };
-
   return (
     <>
```
Comment (rodgobbi) [inline frontend/src/components/RecommendationForm.jsx]: Also remove the React Fragment here.

**Review by rodgobbi — APPROVED** [review]
Cool, I like `Formik`  it's a good abstraction that works for many forms.

File: frontend/src/components/SignUp.jsx (hunk 1 above)
Comment (Moomia) [reply frontend/src/components/SignUp.jsx]: Oh indeed! Somehow it didn't occur to me that the `Formik` tag wraps everything into one single element 😅

**Pull request merged**